sensor/common/resources/metadatagetter/datastore_impl.go

```diff
@@ -0,0 +1,89 @@
+package metadatagetter
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// NamespaceAnnotationStore represents the functions needed to resolve notification metadata.
+type NamespaceAnnotationStore interface {
```

Comment:
We defined this interface as Metadata getter in pkg - let's use that

Reply:
Metadata getter in pkg defines the `GetAnnotationValue` function. The purpose of this interface is to resolve a namespace name to it's list of annotations as is done in the workflow for Sensor and Admission Controller. When we refactor Admission Controller to utilize this common code, all we will need to do is define the Singleton in the Admission Controller location to use its implementation of Namespace Store, since it will also implement this interface

Reply:
We'll have  to move this interface also out to pkg at some point then correct?